internal/parsers/hclparser/hcltype.go

```diff
@@ -8,48 +8,50 @@ import (
 	"github.com/hashicorp/hcl/v2/hclsyntax"
 	"github.com/mineiros-io/terradoc/internal/entities"
 	"github.com/mineiros-io/terradoc/internal/types"
-	"github.com/zclconf/go-cty/cty"
-	"github.com/zclconf/go-cty/cty/function"
-	"github.com/zclconf/go-cty/cty/gocty"
 )
 
+// TODO: refactor - the next two functions are identical apart from the types slice that they use
 func GetVarTypeFromExpression(expr hcl.Expression) (entities.Type, error) {
-	return getTypeFromExpression(expr, varFunctions())
-}
+	t, err := processType(expr, types.TerraformEmptyType)
+	if err != nil {
+		return entities.Type{}, err
+	}
 
-func GetOutputTypeFromExpression(expr hcl.Expression) (entities.Type, error) {
-	return getTypeFromExpression(expr, outputFunctions())
+	found := false
+	for _, tt := range types.VariableTypes {
+		if t.TFType == tt {
```

Comment:
Maybe I missed something, but can't you just return as soon as you find the type ? then you don't need the found boolean (if the loop ends then you return as error, not found, etc).